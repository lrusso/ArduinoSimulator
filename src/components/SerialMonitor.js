import React from "react"
import SerialMonitorLabel from "./SerialMonitorLabel"
import SerialMonitorInput from "./SerialMonitorInput"
import SerialMonitorButton from "./SerialMonitorButton"
import SerialMonitorData from "./SerialMonitorData"
import { sendSerialData } from "../utils/interpreter"
import { useSimulatorContext } from "../contexts/SimulatorContext"

const SerialMonitor = () => {
  const { simulatorRunning } = useSimulatorContext()
  const [inputValue, setInputValue] = React.useState("")

  const _sendSerialData = () => {
    sendSerialData(inputValue)
    setInputValue("")
  }

  React.useEffect(() => {
    if (!simulatorRunning) {
      setInputValue("")
    }
  }, [simulatorRunning])

  return (
    <div style={styles.container}>
      <SerialMonitorLabel />
      <div style={styles.input}>
        <SerialMonitorInput
          sendSerialData={_sendSerialData}
          value={inputValue}
          onChange={setInputValue}
          simulatorRunning={simulatorRunning}
        />
        <SerialMonitorButton
          sendSerialData={_sendSerialData}
          simulatorRunning={simulatorRunning}
        />
      </div>
      <SerialMonitorData />
    </div>
  )
}

const styles = {
  container: {
    backgroundColor: "#F2F2F2",
    borderBottom: "1px solid #D3D3D3",
  },
  input: {
    width: "100%",
    display: "flex",
  },
}

export default SerialMonitor
