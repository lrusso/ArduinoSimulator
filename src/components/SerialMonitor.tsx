import React from "react"
import SerialMonitorLabel from "./SerialMonitorLabel"
import SerialMonitorInput from "./SerialMonitorInput"
import SerialMonitorButton from "./SerialMonitorButton"
import SerialMonitorData from "./SerialMonitorData"
import { sendSerialData } from "../utils/interpreter"
import { useSimulatorContext } from "../contexts/SimulatorContext"

const SerialMonitor = () => {
  const { simulatorRunning } = useSimulatorContext()
  const [inputValue, setInputValue] = React.useState<string>("")

  const _sendSerialData = () => {
    sendSerialData(inputValue)
    setInputValue("")
  }

  React.useEffect(() => {
    if (!simulatorRunning) {
      setInputValue("")
    }
  }, [simulatorRunning])

  return (
    <div style={styles.container}>
      <SerialMonitorLabel />
      <div style={styles.input}>
        <SerialMonitorInput
          sendSerialData={_sendSerialData}
          value={inputValue}
          onChange={setInputValue}
          simulatorRunning={simulatorRunning}
        />
        <SerialMonitorButton
          sendSerialData={_sendSerialData}
          simulatorRunning={simulatorRunning}
        />
      </div>
      <SerialMonitorData />
    </div>
  )
}

const styles: { [key: string]: React.CSSProperties } = {
  container: {
    backgroundColor: "#181818",   
  },
  input: {
    width: "100%",
    display: "flex",
  },
}

export default SerialMonitor
