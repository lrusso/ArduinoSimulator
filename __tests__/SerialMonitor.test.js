import React from "react"
import renderer from "react-test-renderer"
import SerialMonitor from "../src/components/SerialMonitor"

describe("SerialMonitor", () => {
  test("Rendering 1", () => {
    const tree = renderer.create(<SerialMonitor />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
