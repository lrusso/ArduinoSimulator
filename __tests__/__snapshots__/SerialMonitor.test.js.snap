// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SerialMonitor Rendering 1 1`] = `
<div
  style={
    {
      "backgroundColor": "#F2F2F2",
      "borderBottom": "1px solid #D3D3D3",
    }
  }
>
  <div
    style={
      {
        "cursor": "default",
        "float": "left",
        "fontFamily": "Arial",
        "fontSize": "13px",
        "fontWeight": "bold",
        "lineHeight": 2,
        "marginLeft": "7px",
        "marginRight": "1px",
      }
    }
  >
    SERIAL MONITOR
  </div>
  <div
    style={
      {
        "display": "flex",
        "width": "100%",
      }
    }
  >
    <input
      onChange={[Function]}
      onKeyUp={[Function]}
      readOnly={true}
      style={
        {
          "WebkitAppearance": "none",
          "backgroundColor": "#D9D9D9",
          "border": "1px solid silver",
          "borderRadius": 0,
          "cursor": "default",
          "flex": 2,
          "fontFamily": "Arial",
          "fontSize": "13px",
          "height": "18px",
          "marginBottom": "7px",
          "marginLeft": "5px",
          "marginRight": "3px",
          "outline": "none",
          "padding": "4px",
        }
      }
      type="text"
      value=""
    />
    <input
      disabled={true}
      onClick={[Function]}
      style={
        {
          "WebkitAppearance": "none",
          "backgroundColor": "#D9D9D9",
          "border": "1px solid silver",
          "borderRadius": 0,
          "color": "gray",
          "fontFamily": "Arial",
          "fontSize": "13px",
          "height": "28px",
          "marginLeft": "2px",
          "marginRight": "5px",
          "width": "99px",
        }
      }
      type="submit"
      value="SEND"
    />
  </div>
  <div
    style={
      {
        "backgroundColor": "white",
        "bottom": 0,
        "height": "137px",
        "left": 0,
        "overflow": "scroll",
        "position": "fixed",
        "right": 0,
      }
    }
  >
    <div
      style={
        {
          "MozUserSelect": "text",
          "WebkitUserSelect": "text",
          "fontFamily": "monospace,monospace",
          "overflowWrap": "break-word",
          "padding": "10px",
          "userSelect": "text",
          "whiteSpace": "pre-line",
        }
      }
    />
  </div>
</div>
`;
